cosmasken | 2024-02-23 11:52:38 UTC | #1

My internet identity integration has been working correctly until Feb 22 .When i run the frontend I get this error The define key "process.env.INTERNET-IDENTITY_CANISTER_ID" contains invalid identifier "INTERNET-IDENTITY_CANISTER_ID" .I haven't changed anything in my code .

-------------------------

Severin | 2024-02-23 12:09:10 UTC | #2

Any chance you accidentally changed a `_` to a `-`? Can you try replacing `INTERNET-IDENTITY_CANISTER_ID` with `INTERNET_IDENTITY_CANISTER_ID`?

-------------------------

frederikrothenberger | 2024-02-23 12:15:33 UTC | #3

Hi @cosmasken 
How are you adding Internet Identity to your local setup? Note: with the latest release, we introduced an issue with `dfx deps pull`. This should be fixed with the next release, to be deployed on Monday.

-------------------------

cosmasken | 2024-02-23 12:46:26 UTC | #4

I'm using internet-identity . Should I change to internet_identity?

-------------------------

cosmasken | 2024-02-23 12:47:29 UTC | #5

I was using the pulled internet identity but had to change when it gave me the error of different hashes for the wasm and in pulled json

-------------------------

Severin | 2024-02-23 12:53:49 UTC | #6

It doesn't matter what you call the canister, but the environment variable uses `_` in place of `-`. ([source code](https://github.com/dfinity/sdk/blob/master/src/dfx/src/lib/builders/mod.rs#L400)) So `process.env.<something>` should never contain a dash

-------------------------

frederikrothenberger | 2024-02-23 13:00:38 UTC | #7

Yes, this is a bug on our side. It will be fixed when [this proposal](https://nns.ic0.app/proposal/?u=qoctq-giaaa-aaaaa-aaaea-cai&proposal=128085) is executed.

-------------------------

